### DISCUSSION TITLE
Codespaces: support signed commits

### DISCUSSION THREAD

**Discussion by fulldecent** [community/community#17223] [Codespaces]
# Tech context

*I am in the limited beta for Codespaces.*

My relevant stack is: macOS / Apple silicon / VS Code / Remote Containers extension / GitHub Codespaces / GPG Keychain (macOS GUI for GPG signing).

And my workflow is to have a local repository (which upstreams onto GitHub) "reopened" through a GitHub Codespace.

# Ethos context

I would like to sign my commits.

Let's assume that I believe one of the 10,000 dependencies in the code I am working on would do BAD THINGS if it was able to sign GPG messages as me.

# What I tried

I tried downloading code locally, opening with the tech stack above and made changes. Run test code remotely, everything is beautiful, thank you.

Now I want to make a commit. So I use the side bar to prepare my commit. But commits don't work in this context.

So my workaround is to use the terminal to `git diff --staged > diff.patch`, download that file, and then `patch -p1 diff.patch` locally and make the commit.

This workaround is great for single commits but it would get old fast if I want to deal with multiple branches or real work.

**Comment by agardnerIT** [comment]
For any else finding this via Google. It is now possible to sign commits via visual studio on codespaces:

<img width="752" alt="Screenshot 2023-05-01 at 12 42 57 pm" src="https://user-images.githubusercontent.com/26523841/235394404-351fc10a-2a0a-40b5-b8a2-167db15718d5.png">
